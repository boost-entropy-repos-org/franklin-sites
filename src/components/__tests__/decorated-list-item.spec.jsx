import React from 'react';
import renderer from 'react-test-renderer';

import DecoratedListItem from '../decorated-list-item';

describe('DecoratedListItem component', () => {
  test('should render', () => {
    const component = renderer
      .create(<DecoratedListItem title="Title">Content</DecoratedListItem>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render without title', () => {
    const component = renderer
      .create(<DecoratedListItem>Content</DecoratedListItem>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render compact', () => {
    const component = renderer
      .create(
        <DecoratedListItem title="Title" compact>
          Content
        </DecoratedListItem>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render compact without title', () => {
    const component = renderer
      .create(<DecoratedListItem compact>Content</DecoratedListItem>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render inline', () => {
    const component = renderer
      .create(<DecoratedListItem inline>Content</DecoratedListItem>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
