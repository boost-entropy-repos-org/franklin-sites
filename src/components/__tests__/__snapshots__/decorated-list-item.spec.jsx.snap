// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecoratedListItem component should render 1`] = `
<div
  className="decorated-list-item"
>
  <div
    className="decorated-list-item__title"
  >
    <h5
      className="bold"
    >
      Title
    </h5>
  </div>
  <div
    className="decorated-list-item__content"
  >
    Content
  </div>
</div>
`;

exports[`DecoratedListItem component should render compact 1`] = `
<div
  className="decorated-list-item decorated-list-item--compact"
>
  <div
    className="decorated-list-item__title"
  >
    <h5
      className="bold"
    >
      Title
    </h5>
  </div>
  <div
    className="decorated-list-item__content"
  >
    Content
  </div>
</div>
`;

exports[`DecoratedListItem component should render compact without title 1`] = `
<div
  className="decorated-list-item decorated-list-item--compact"
>
  <div
    className="decorated-list-item__title"
  >
    <h5
      className="bold"
    />
  </div>
  <div
    className="decorated-list-item__content"
  >
    Content
  </div>
</div>
`;

exports[`DecoratedListItem component should render inline 1`] = `
<div
  className="decorated-list-item decorated-list-item--inline"
>
  <div
    className="decorated-list-item__title"
  >
    <h5
      className="bold"
    />
  </div>
  <div
    className="decorated-list-item__content"
  >
    Content
  </div>
</div>
`;

exports[`DecoratedListItem component should render without title 1`] = `
<div
  className="decorated-list-item"
>
  <div
    className="decorated-list-item__title"
  >
    <h5
      className="bold"
    />
  </div>
  <div
    className="decorated-list-item__content"
  >
    Content
  </div>
</div>
`;
